src/Forms/GridField/GridField.php

```diff
@@ -521,6 +523,28 @@ public function FieldHolder($properties = [])
     {
         $this->extend('onBeforeRenderHolder', $this, $properties);
 
+        // Set GridField to read-only if sudo mode is required for the DataObject being managed
+        // and sudo mode is not active
+        $modelClass = null;
+        try {
+            $modelClass = $this->getModelClass();
+        } catch (LogicException) {
+            // noop
+        }
```

Comment:
What scenario did you have where a model class wasn't defined by this stage? That shouldn't happen unless something is misconfigured.

Reply:
1) SilverStripe\Forms\Tests\GridField\GridFieldTest::testGridFieldCustomFragments
LogicException: GridField doesn't have a modelClassName, so it doesn't know the columns of this grid.

https://github.com/creative-commoners/recipe-kitchen-sink/actions/runs/13359908314/job/37307887025#step:12:211

$this->getModelClass() can throw and exception so we need to handle it

Reply:
> $this->getModelClass() can throw an exception so we need to handle it

We throw a lot of exceptions explicitly to _not_ be caught by our code, so that developers can be aware of misconfigurations or logic errors in their code. So just because we're throwing an exception doesn't mean we need to catch it.

That said, it looks like that test is (inadvertently) showing that GridField can be used in a way that doesn't require showing columns from the objects held in its list - which would be an _extreme_ edge case but I guess there's no reason to break that flexibility here.

I'll add a suggestion to update the `noop` comment so I don't see this in the future and freak out lol